streambed-logged/src/lib.rs

```diff
@@ -605,6 +655,53 @@ fn find_offset(
     }
 }
 
+fn recover_active_file(
+    topic_file_op: &mut TopicFileOp,
+    read_buffer_size: usize,
+    max_record_size: usize,
+) -> Result<(), TopicFileOpError> {
+    let mut open_options = OpenOptions::new();
+    open_options.read(true).write(true);
+    let mut topic_file = topic_file_op.open_active_file(open_options)?;
+    let mut buf = BytesMut::with_capacity(read_buffer_size);
+    let mut decoder = StorableRecordDecoder::new(max_record_size);
+    let mut bytes_read = None;
+    loop {
+        let Ok(len) = read_buf(&mut topic_file, &mut buf) else {
+                break;
+            };
+
+        let before_decode_len = buf.len();
+
+        let decode_fn = if len == 0 {
```

Comment:
This looks like a bug here.  Should be in the loop.

Reply:
If it meant to be like this?  (Not sure) 
```rust
        while let Ok(Some(_)) = if buf.len() == 0 {
            StorableRecordDecoder::decode_eof(&mut decoder, &mut buf)
        } else {
            StorableRecordDecoder::decode(&mut decoder, &mut buf)
        } {}

```

Reply:
ah yeah... good catch

Reply:
Wait... the `len` returned by `read_buf` is from `File::read`... not the buffer itself. We should be good.

Reply:
the len return by read is not relevant is it?

Reply:
Yes, it is relevant. A file read returning 0 indicates EOF.

Reply:
OK I looked more closely and I can see how it works.   No bug here.  (I think the code is a bit awkward but part of that seems to be inherited from tokio.)